### DISCUSSION TITLE
Failed to write previous plot error: 28

### DISCUSSION THREAD

**Discussion by digitalspaceport** [Chia-Network/bladebit#124] [General]
Fantastic work with this Harold! Just getting around to doing some in depth tests on it and ran into this issue on plot 3 creation. BB 1.2 Ubuntu 20.04.3 LTS

Failed to write previous plot file /mnt/plots/plot-k32-2021-10-XXXXX with error: 28

Brief on system storage is 4x 1TB ssd's in RAID0 btrfs. Usually very performant drives even outside the drives cache.

Here is a cat of directory. It looks like it was still in copy of plot 2 when it bit the dust. It could be the HBA I have in there not playing nice with discard, current card is H310 not flashed. It is running in IR mode with drives not configured inside the perc.

Will swap with H710 flashed which does handle discard. FS is btrfs so continuous trim should be supported.

-rw-r----- 1 root root 108800180224 Oct  8 16:51 plot-1
-rw-r----- 1 root root  83101474816  Oct  8 17:09 plot-2
-rw-r----- 1 root root                      0  Oct  8 17:07 plot-3 < crash

**Comment by harold-b** [comment]
Hmm, well error 28 is [ENOSPC  (No space left on device)](https://man7.org/linux/man-pages/man3/errno.3.html). From bladebit's POV it can't do anything about that, since that's the error code returned from a `write` operation. I would assume there's something going on in another layer where it's eating up space... Perhaps there's some left-over deleted files from zombie processes?

**Reply by digitalspaceport** [reply]
This was it. Checked the dir I had mounted and it had a .dot dir hanging out with a bunch of temp files. Should have checked df -h before posting. Thanks!

**Reply by harold-b** [reply]
No prob! Happy to help
